### PR TITLE
Support for a custom require function

### PR DESCRIPTION
**Pull request by generalov**
The PR to use a custom loader for the commands. Actually all necessary functionality already implemented internally in the Yargs. Thank you guys for the great tool. :)

For example we can use [@std/esm](https://www.npmjs.com/package/@std/esm) as a loader to load the commands written in ES6 modules syntax.
```
const loader = require('@std/esm')(module)
const yargs = require('yargs')

yargs(process.argv.slice(2), undefined, loader)
  .commandDir('cmds', {
    extensions: [ 'js', 'mjs' ]
  })
  .demandCommand()
  .argv
```
cmds/es6.mjs
```
import fs from 'fs'

export const command = 'esm'
export const description = 'The commands can be defined in ES6 modules'

export function handler (argv) {
  console.log("I'm ES6 command!")
}
```

### CODE REVIEW

**Comment by bcoe** [comment]
@generalov this seems like a reasonable addition; would it be possible to add a test?

File: index.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ Argv(process.argv.slice(2))
 
 module.exports = Argv
 
-function Argv (processArgs, cwd) {
-  const argv = yargs(processArgs, cwd, require)
+function Argv (processArgs, cwd, parentRequire) {
```
Comment (bcoe) [inline index.js]: I'm thinking maybe we make the second parameter take an object:

```js
{cwd: 'foo', require: func () {}}
```

if a string is provided we'd fallback to the `cwd` behavior.

**Comment by bcoe** [comment]
@generalov do my comments make sense, would love to see this over the finish line soon.

**Comment by generalov** [comment]
@bcoe my apologues for the delay. Could you please advise me that is the best place to put a test?

**Comment by bcoe** [comment]
@generalov it's my turn to apologize for taking a long time to reply! I've been busy with a project at work, and another codebase I've been coming up to speed on.

I would add a test in `test/yargs.js` -- perhaps pull in `@std/esm` as a dev dependency, and you can add a unit test for a real-world example?

**Comment by bcoe** [comment]
@generalov just checking in to see if you're still interested in landing this feature 👍

**Comment by bcoe** [comment]
@genepaul sorry to keep prodding you, but would love to get a unit test added so that we can land this; or perhaps someone else would like to see this over the finish line?

**Comment by bcoe** [comment]
@generalov hey, given that this has been inactive for a while I'm going to go ahead and close this issue ... one thought, if you do need this functionality, is that you could simply use:

```js
const yargs = require('yargs/yargs')
yargs(processArgs, cwd, parentRequire || require).argv
```

if you need this functionality. I'd be happy to revisit this feature if you do want to make a case.

**Pull request closed without merging**